helpers/Delivery.php

```diff
@@ -0,0 +1,288 @@
+use \common_session_SessionManager;
+
+class Delivery extends Proctoring
+{
+    /**
+     * Gets a list of available deliveries for a test site
+     *
+     * @param string $testCenter
+     * @param array [$options]
+     * @return array
+     * @throws ServiceNotFoundException
+     * @throws \common_Exception
+     * @throws \common_exception_Error
+     */
+    public static function getDeliveries($testCenter)
+    {
+        $currentUser = common_session_SessionManager::getSession()->getUser();
+        $deliveryService = ServiceManager::getServiceManager()->get('taoProctoring/delivery');
+        $deliveries = $deliveryService->getProctorableDeliveries($currentUser);
+
+        $entries = array();
+
+        $all = array(
+            'id' => 'all',
+            'url' => _url('monitoringAll', 'Delivery', null, array('testCenter' => $testCenter->getUri())),
+            'label' => __('All Deliveries'),
+            'cls' => 'dark',
+            'stats' => array(
+                'awaitingApproval' => 0,
+                'inProgress' => 0,
+                'paused' => 0
+            )
+        );
+
+        $mocks = array(
+            array(
+                'stats' => array(
+                    'awaitingApproval' => 0,
+                    'inProgress' => 0,
+                    'paused' => 0
+                ),
+                'properties' => array(),
+            ),
+            array(
+                'stats' => array(
+                    'awaitingApproval' => 3,
+                    'inProgress' => 32,
+                    'paused' => 12
+                ),
+                'properties' => array(
+                    'periodStart' => '2015-11-09 00:00',
+                    'periodEnd' => '2015-11-17 09:20'
+                )
+            ),
+            array(
+                'stats' => array(
+                    'awaitingApproval' => 0,
+                    'inProgress' => 15,
+                    'paused' => 1
+                ),
+                'properties' => array(
+                    'periodStart' => '2015-11-09 00:00',
+                    'periodEnd' => '2015-11-17 09:20'
+                )
+            ),
+            array(
+                'stats' => array(
+                    'awaitingApproval' => 1,
+                    'inProgress' => 10,
+                    'paused' => 8
+                ),
+                'properties' => array(
+                    'periodStart' => '2015-11-09 00:00',
+                    'periodEnd' => '2015-11-17 09:20'
+                )
+            ),
+        );
+
+        foreach ($deliveries as $delivery) {
+            $entries[] = array_merge(array(
+                'id' => $delivery->getUri(),
+                'url' => _url('monitoring', 'Delivery', null, array('delivery' => $delivery->getUri(), 'testCenter' => $testCenter->getUri())),
+                'label' => $delivery->getLabel(),
+                'text' => __('Monitor')
+            ), $mocks[array_rand($mocks)]);
```

Comment:
actually, I would recommend not randomise this information but take the mocks data in order. the first delivery will *always* have the mock data 1 and the second delivery the mock number 2, etc. If there is too much deliveries, reiterate with the first mock... indeed, I am really affraid that the client will try to play a lot with it and be confused about the changing states after refresh. It is fine to randomize, but it is crucial to be consistent. Mocks are suppose to simulate a behaviour as close as the reality as possible after all.
 This comment is also applicable for delivery/index, delivery/monitoring and report/index.